experiments/ClimaEarth/setup_run.jl

```diff
@@ -687,37 +627,56 @@ function setup_and_run(config_dict::AbstractDict)
     We use the `ClimaComms.@elapsed` macro to time the simulation on both CPU and GPU, and use this
     value to calculate the simulated years per day (SYPD) of the simulation.
     =#
-    walltime = ClimaComms.@elapsed comms_ctx.device begin
+    @info "Starting coupling loop"
+    walltime = ClimaComms.@elapsed cs.comms_ctx.device begin
         s = CA.@timed_str begin
-            solve_coupler!(cs)
+            while cs.t[] <= cs.tspan[end]
+                step!(cs)
+            end
         end
     end
-    @info(walltime)
+    @info "Simulation took $(walltime) seconds"
 
-    ## Use ClimaAtmos calculation to show the simulated years per day of the simulation (SYPD)
-    es = CA.EfficiencyStats((tspan[1], tspan[2]), walltime)
-    sypd = CA.simulated_years_per_day(es)
-    n_atmos_steps = atmos_sim.integrator.step
-    walltime_per_atmos_step = es.walltime / n_atmos_steps
+    simulated_seconds_per_second = float(cs.tspan[end] - cs.tspan[begin]) / walltime
+    simulated_years_per_day = simulated_seconds_per_second / (365.25 * walltime)
+    sypd = simulated_years_per_day
+    n_atmos_steps = cs.model_sims.atmos_sim.integrator.step
+    walltime_per_atmos_step = walltime / n_atmos_steps
```

Comment:
I think we should instead get the walltime per coupled step here. Of course atmos will dominate the time taken, but it's not the coupler's responsibility to track atmos performance. The only reason I can think we might want to keep it as walltime per atmos step would be for more direct comparisons to ClimaAtmos standalone runs

Reply:
I think the reason was here was ineed to compare with Atmos standalone. I'll turn it into coupling step. I agree that it makes more sense.